src/main/java/net/glowstone/entity/passive/GlowChestedHorse.java

```diff
@@ -18,6 +33,14 @@ public boolean isCarryingChest() {
 
     @Override
     public void setCarryingChest(boolean carryingChest) {
-        metadata.set(MetadataIndex.CHESTED_HORSE_HAS_CHEST, carryingChest);
+        if (carryingChest != isCarryingChest()) {
+            metadata.set(MetadataIndex.CHESTED_HORSE_HAS_CHEST, carryingChest);
+            createNewInventory();
+        }
     }
+
+    /**
+     * Creates and sets a new inventory, and copies equipment over from the existing inventory.
+     */
+    protected abstract void createNewInventory();
```

Comment:
I think this should return `<InventoryT>`, and the constructor should set the `inventory` field to the return value. The method implementations shouldn't have to update the field imo.